app/src/main/java/com/github/se/wanderpals/ui/screens/trip/finance/CreateExpense.kt

```diff
@@ -0,0 +1,359 @@
+  var selectedMenu2 by remember { mutableStateOf("") }
+  var expenseTitle by remember { mutableStateOf("") }
+  var expenseAmount by remember { mutableStateOf("") }
+  var expenseDate by remember { mutableStateOf("") }
+  var showDatePicker: Boolean by remember { mutableStateOf(false) }
+
+  var errorText by remember { mutableStateOf("") }
+
+  val users by viewModel.users.collectAsState()
+
+  var checkboxes = remember { mutableStateListOf<Boolean>() }
+
+  if (checkboxes.size != users.size) {
+    checkboxes = users.map { false }.toMutableStateList()
+  }
+
+  WanderPalsTheme {
+    Scaffold(
+        topBar = {
+          TopAppBar(
+              title = {
+                Text(
+                    text = "Add an expense",
+                    style = MaterialTheme.typography.titleLarge,
+                    modifier = Modifier.testTag("createExpenseTitle"))
+              },
+              navigationIcon = {
+                IconButton(
+                    onClick = { navActions.goBack() }, modifier = Modifier.testTag("BackButton")) {
+                      Icon(
+                          imageVector = Icons.AutoMirrored.Filled.ArrowBack,
+                          contentDescription = "Go back",
+                      )
+                    }
+              },
+              colors =
+                  TopAppBarDefaults.topAppBarColors(
+                      containerColor = MaterialTheme.colorScheme.primaryContainer,
+                  ),
+          )
+        }) { paddingValues ->
+          Box(modifier = Modifier.padding(paddingValues).fillMaxSize().background(Color.White)) {
+            Box(modifier = Modifier.verticalScroll(rememberScrollState(), true)) {
+              Column(modifier = Modifier.testTag("createExpenseContent").background(Color.White)) {
+                HorizontalDivider(color = MaterialTheme.colorScheme.outline)
+
+                Spacer(modifier = Modifier.padding(4.dp))
+
+                // Content
+                OutlinedTextField(
+                    value = expenseTitle,
+                    onValueChange = { if (it.length <= 50) expenseTitle = it else {} },
+                    label = { Text("Expense title") },
+                    placeholder = { Text("Give a name to your expense") },
+                    modifier =
+                        Modifier.testTag("expenseTitle")
+                            .padding(horizontal = 24.dp, vertical = 8.dp)
+                            .fillMaxWidth(),
+                    singleLine = true,
+                    suffix = {
+                      Text(
+                          text = "${expenseTitle.length}/50",
+                          style = MaterialTheme.typography.bodyMedium)
+                    },
+                    isError = errorText.isNotEmpty(),
+                )
+
+                OutlinedTextField(
+                    value = expenseAmount,
+                    onValueChange = { if (it.length <= 20) expenseAmount = it else {} },
```

Comment:
Same as before, you can probably remove the empty `else` statement

Reply:
Correct, I often get an error if I don't put an else in this kind of case but here it's a Unit so there is no problem.